### PR TITLE
docs: predicate and script deploys

### LINKED ISSUE #3263: Predicate and script deploy follow-ups
> [!IMPORTANT]  
> This must go after #3251 and not be included in release #3250 

- [x] The new tests added for docs are currently failing, this is because we require deploy in pretest

### PR DESCRIPTION
**Pull request by danielbate**
- Closes #3263

# Release notes

In this release, we:

- Added documentation for deploying scripts and predicates

# Summary

- Adds a new pretest script for `docs-snippets` so we can execute `fuels deploy` for loader bytecode
- Adds docs and test for script and predicate deploys

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: apps/docs/src/guide/predicates/deploying-predicates.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Deploying Predicates
+
+In order to optimize the cost of your recurring predicate executions, we recommend first deploying your predicate. This can be done using the [Fuels CLI](../fuels-cli/index.md) and running the [deploy command](../fuels-cli/commands#fuels-deploy).
+
+By deploying the predicate, it's bytecode is stored on chain as a blob. The SDK will then produce bytecode that can load the blob on demand that can execute the predicate. This far reduces the repeat execution cost of the predicate.
```
Comment (nedsalk) [inline apps/docs/src/guide/predicates/deploying-predicates.md:5]:
```suggestion
By deploying the predicate, its bytecode is stored on chain as a blob. The SDK will then produce bytecode that can load the blob on demand to execute the original predicate. This far reduces the repeat execution cost of the predicate.
```

File: apps/docs/src/guide/predicates/deploying-predicates.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+## How to Deploy a Predicate
+
+To deploy a predicate, we can use the [Fuels CLI](../fuels-cli/index.md) and execute the [deploy command](../fuels-cli/commands#fuels-deploy).
+
+This will perform the following actions:
+
+1. Compile the script using your `forc` version
+1. Deploys the built script binary to the chain as a blob
+1. Generates a script that loads the blob that can be used to execute the script
+1. Generates types for both the script and the loader that you can use in your application
```
Comment (nedsalk) [inline apps/docs/src/guide/predicates/deploying-predicates.md:16]:
```suggestion
1. Compile the predicate using your `forc` version
1. Deploy the built predicate binary to the chain as a blob
1. Generates a new, smaller predicate that loads the deployed predicate's blob
1. Generates types for both the predicate and the loader that you can use in your application
```

File: apps/docs/src/guide/scripts/deploying-scripts.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Deploying Scripts
+
+In order to optimize the cost of your recurring script executions, we recommend first deploying your script. This can be done using the [Fuels CLI](../fuels-cli/index.md) and running the [deploy command](../fuels-cli/commands#fuels-deploy).
+
+By deploying the script, it's bytecode is stored on chain as a blob. The SDK will then produce bytecode that can load the blob on demand that can execute the script. This far reduces the repeat execution cost of the script.
```
Comment (nedsalk) [inline apps/docs/src/guide/scripts/deploying-scripts.md:5]:
```suggestion
By deploying the script, its bytecode is stored on chain as a blob. The SDK will then produce bytecode that can load the blob on demand to execute the original script. This far reduces the repeat execution cost of the script.
```

File: apps/docs/src/guide/scripts/deploying-scripts.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+## How to Deploy a Script
+
+To deploy a script, we can use the [Fuels CLI](../fuels-cli/index.md) and execute the [deploy command](../fuels-cli/commands#fuels-deploy).
+
+This will perform the following actions:
+
+1. Compile the script using your `forc` version
+1. Deploys the built script binary to the chain as a blob
+1. Generates a script that loads the blob that can be used to execute the script
+1. Generates types for both the script and the loader that you can use in your application
```
Comment (nedsalk) [inline apps/docs/src/guide/scripts/deploying-scripts.md:16]:
```suggestion
1. Compile the script using your `forc` version
1. Deploy the built script binary to the chain as a blob
1. Generate a script that loads the blob that can be used to execute the script
1. Generate types for both the script and the loader that you can use in your application
```

**Review by nedsalk — DISMISSED** [review]

**Comment by petertonysmith94** [comment]
Back in review until ready.

File: apps/docs/src/guide/predicates/deploying-predicates.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Deploying Predicates
+
+In order to optimize the cost of your recurring predicate executions, we recommend first deploying your predicate. This can be done using the [Fuels CLI](../fuels-cli/index.md) and running the [deploy command](../fuels-cli/commands#fuels-deploy).
+
+By deploying the predicate, its bytecode is stored on chain as a blob. The SDK will then produce bytecode that can load the blob on demand to execute the original predicate. This far reduces the repeat execution cost of the predicate.
+
+## How to Deploy a Predicate
+
+To deploy a predicate, we can use the [Fuels CLI](../fuels-cli/index.md) and execute the [deploy command](../fuels-cli/commands#fuels-deploy).
```
Comment (petertonysmith94) [inline apps/docs/src/guide/predicates/deploying-predicates.md:9]: I think we should had relevant docs in the `fuels-deploy` command. Same as what we did with proxy.

File: apps/docs/src/guide/scripts/deploying-scripts.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Deploying Scripts
+
+In order to optimize the cost of your recurring script executions, we recommend first deploying your script. This can be done using the [Fuels CLI](../fuels-cli/index.md) and running the [deploy command](../fuels-cli/commands#fuels-deploy).
+
+By deploying the script, its bytecode is stored on chain as a blob. The SDK will then produce bytecode that can load the blob on demand to execute the original script. This far reduces the repeat execution cost of the script.
+
+## How to Deploy a Script
+
+To deploy a script, we can use the [Fuels CLI](../fuels-cli/index.md) and execute the [deploy command](../fuels-cli/commands#fuels-deploy).
```
Comment (petertonysmith94) [inline apps/docs/src/guide/scripts/deploying-scripts.md:9]: I think we should had relevant docs in the `fuels-deploy` command. Same as what we did with proxy.

File: apps/docs-snippets/src/guide/scripts/deploying-scripts.test.ts
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/scripts/deploying-scripts.test.ts:1]: image

**Review by Torres-ssf — DISMISSED** [review]

File: apps/docs-snippets/src/guide/scripts/deploying-scripts.test.ts [hunk 7]
```diff
@@ -61,13 +48,13 @@ describe.skip('Deploying Scripts', () => {
     const { value, gasUsed } = await waitForResult();
     // #endregion deploying-scripts
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/scripts/deploying-scripts.test.ts:49]:
```suggestion
    // value: 10
    // #endregion deploying-scripts
```

File: apps/docs/src/guide/predicates/deploying-predicates.md (hunk 5 above)
Comment (petertonysmith94) [reply apps/docs/src/guide/predicates/deploying-predicates.md:9]: fuels-cli/commands#fuels-deploy

**Review by petertonysmith94 — DISMISSED** [review]

File: .changeset/thirty-parrots-sniff.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+"fuels": patch
```
Comment (petertonysmith94) [inline .changeset/thirty-parrots-sniff.md:2]:
```suggestion
```

**Review by petertonysmith94 — DISMISSED** [review]

**Review by nedsalk — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]

**Review by nedsalk — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**